resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/UserAdapter.java

```diff
@@ -0,0 +1,249 @@
+            case UserModel.FIRST_NAME:
+                entity.setFirstName(value);
+                break;
+            case UserModel.LAST_NAME:
+                entity.setLastName(value);
+                break;
+            case "created_timestamp":
+                entity.setCreatedTimestamp(value == null ? null : Long.valueOf(value));
+                break;
+            case "super_user":
+                entity.setSuperUser(value == null ? null : Integer.valueOf(value));
+                break;
+            case "organizations":
+                entity.setOrganizations(OrganizationObject.listFromString(value));
+                break;
+            default:
+                super.setSingleAttribute(name, value);
+        }
+        save();
+    }
+
+    @Override
+    public void removeAttribute(String name) {
+        log.debug("removeAttribute: " + name);
+        switch (name) {
+            case UserModel.USERNAME:
+                entity.setEmail(null);
+                break;
+            case UserModel.EMAIL:
+                entity.setEmail(null);
+                break;
+            case UserModel.FIRST_NAME:
+                entity.setFirstName(null);
+                break;
+            case UserModel.LAST_NAME:
+                entity.setLastName(null);
+                break;
+            case "created_timestamp":
+                entity.setCreatedTimestamp(null);
+                break;
+            case "super_user":
+                entity.setSuperUser(null);
+                break;
+            case "organizations":
+                entity.setOrganizations(null);
+                break;
+            default:
+                super.removeAttribute(name);
+        }
+        save();
+    }
+
+    @Override
+    public void setAttribute(String name, List<String> values) {
+        log.debug("setAttribute: " + name);
+        if (values == null || values.isEmpty()) {
+            removeAttribute(name);
+        } else {
+            setSingleAttribute(name, values.get(0));
+        }
+    }
+
+    @Override
+    public String getFirstAttribute(String name) {
+        log.debug("getFirstAttribute: " + name);
+        switch (name) {
+            case UserModel.USERNAME:
+                return entity.getEmail();
+            case UserModel.EMAIL:
+                return entity.getEmail();
+            case UserModel.FIRST_NAME:
+                return entity.getFirstName();
+            case UserModel.LAST_NAME:
+                return entity.getLastName();
+            case "created_timestamp":
+                return String.valueOf(entity.getCreatedTimestamp());
+            case "super_user":
+                return String.valueOf(entity.getSuperUser());
+            case "organizations":
+                return entity.getOrganizations() != null ? OrganizationObject.toStringList(entity.getOrganizations())
+                        : "[]";
+            default:
+                return super.getFirstAttribute(name);
+        }
```

Comment:
suggestion(non-blocking): same as the `getAttributeStream` suggesiton below
```suggestion
        return switch (name) {
            case UserModel.USERNAME, UserModel.EMAIL -> entity.getEmail();
            case UserModel.FIRST_NAME -> entity.getFirstName();
            case UserModel.LAST_NAME -> entity.getLastName();
            case "created_timestamp" -> String.valueOf(entity.getCreatedTimestamp());
            case "super_user" -> String.valueOf(entity.getSuperUser());
            case "organizations" ->
                    entity.getOrganizations() != null ? OrganizationObject.toStringList(entity.getOrganizations())
                            : "[]";
            default -> super.getFirstAttribute(name);
        };
```

Reply:
I like this syntax a LOT more, updated this and 3 other switch/case statements to match